database/arango/arango.js

```diff
@@ -0,0 +1,78 @@
+require("dotenv").config();
+const { Database } = require("arangojs");
+const { sendFetch } = require("../../lib/sendFetch");
+const logger = require("./../../lib/log")(__filename);
+
+const arangoModule = {};
+let db;
+
+arangoModule.checkIfDatabaseExists = async (name) => {
+  // returns array of objects containing a '_name' key.
+  // Ex [{ _name: 'lol' }, { _name: 'cakes' }]
+  const names = await db.databases();
+  return names.map((db) => db._name).includes(name);
```

Comment:
It seems that this function requests all of the data from the database in JSON form and then search for the name. I think this can cause some inefficiency. Just like postgres and elasticsearch, I assume that there is a way to query the databases with specific name using arango.

Reply:
Ah yes there is, I considered doing it that way, but eventually concluded that it is better to stick to using their arangojs API.

Another way I could do this is by making a request to a http REST api that arango provides. This is how deleting an account is done -- I use their http REST API. Here are the pros and cons of using each method to check if a user exists:

# Using http REST api:
* Pros
    1) Queries for only one user, so when the promise is received in `const names = await db.databases()`, I don't have to use `map` and `includes` functions.
* Cons
    1) Have to make 2 requests to an arango database -- the first time to get a jwt token, the second time to check if a user exists. This is a type of additional logic that arangojs api would not need.

# Using arangojs api:
* Pros
    1) Only 1 request is made to an arango database -- getting a jwt is not necessary because we authenticated ourselves in the function to connect to the arango database.
    2) Keeps the code consistent - the functions to stop and start a connection to an arango database are done using their arangojs api, so using their arangojs api to check if a user exists is appropriate. 
* Cons
    1) Queries for more than one user, so additional logic must be performed to check if a user exists, after receiving the promise (`using .map and .includes`). Using http REST api, I would not need this type of additional logic.


# Efficiency
I believe there is no noticeable difference in efficiency between the two methods, unless learndatabases.dev gets a looot of users. You can [see here](https://www.arangodb.com/2018/02/nosql-performance-benchmark-2018-mongodb-postgresql-orientdb-neo4j-arangodb/) that on average, it took arango only `1.07 seconds` to aggregate a collection of `1,632,803` documents. Since efficiency is not a concern, I think using arangojs is better for now. In the future, if efficiency becomes a concern, then we can switch to using the http REST api that arango provides.

Reply:
As you said it won't be a big problem for now since we don't have a lot of requests but I want our app to be open to a chance to be so.

I have done some research and found something that can be a clue.

1. https://arangodb.github.io/arangojs/latest/classes/_database_.database.html#query
This function let us use AQL
2. https://www.arangodb.com/docs/stable/aql/tutorial-filter.html
This is how to get document that matches specific condition

So in this case I assume it should be like this:
```
const dbExists await !!db.query(aql`
  FOR d IN database
  FILTER d._name == ${name}
  RETURN d.name
`);
return dbExists
```

Reply:
I tried looking into using `AQL` commands to do user management, but I wasn't able to do so. This is why I used http rest API to delete arango users, because I couldn't use `AQL` commands to do so. I think it is possible for us to make a collection of users to query from, but that would make our code more complex than it has to be, and so we should not do that.

Reply:
https://github.com/garageScript/databases/issues/237